lib/facter/resolvers/containers.rb

```diff
@@ -15,7 +15,17 @@ class << self
 
         def post_resolve(fact_name, _options)
           @fact_list.fetch(fact_name) do
-            read_environ(fact_name) || read_cgroup(fact_name)
+            environ_result = read_environ(fact_name)
+            cgroup_result = read_cgroup(fact_name)
+
+            if environ_result.nil? && cgroup_result.nil?
+              @fact_list[:vm] = 'container_other'
```

Comment:
`VirtualDetector` assumes this resolver will return nil if it doesn't recognize the container type (or if there is none), so that the detector can try other methods. See my comments in https://github.com/puppetlabs/facter/issues/2784#issuecomment-2620234191